src/module.cpp

```diff
@@ -5609,20 +5616,17 @@ RedisModuleTimerID RM_CreateTimer(RedisModuleCtx *ctx, mstime_t period, RedisMod
         if (memcmp(ri.key,&key,sizeof(key)) == 0) {
             /* This is the first key, we need to re-install the timer according
              * to the just added event. */
-            aePostFunction(g_pserver->rgthreadvar[IDX_EVENT_LOOP_MAIN].el, [&]{
+            aePostFunction(g_pserver->rgthreadvar[IDX_EVENT_LOOP_MAIN].el, [period]{
```

Comment:
I'm not seeing how you can get away with doing this at such a low level without a race.  Don't you need *everything* including the RAX search and the adding and removing of timers to all happen as a single event in the context of the AE Lock?